enumcover.go

```diff
@@ -80,29 +81,19 @@ func checkConsts(pass *analysis.Pass, n ast.Node, typeName string) {
 				case *ast.BasicLit:
 					namesForType[unquote(n.Value)] = true
 				case *ast.Ident:
-					if n.Obj != nil {
-						if n.Obj.Kind == ast.Con {
-							if decl, ok := n.Obj.Decl.(*ast.ValueSpec); ok {
-								for _, value := range decl.Values {
-									if lit, ok := value.(*ast.BasicLit); ok {
-										namesForType[unquote(lit.Value)] = true
-									}
-								}
-							}
-						}
-					}
-					namesForType[n.Name] = true
+					namedConst := allConsts[n.Name]
+					namesForType[namedConst.val] = true
 				}
 			}
 		}
 		return true
 	})
-	allConsts := allConstsWithType(pass, typeName)
+
 	if len(allConsts) == 0 {
 		reportNodef(pass, n, "No consts found for type %v", typeName)
 	}
 	for _, want := range allConsts {
-		if !namesForType[want.name] && !namesForType[want.val] {
+		if !namesForType[want.val] {
```

Comment:
There won't also be need for the name part of this check, since all our `namesForType` entries are currently `val` indexed and not a combination of both.